Add Support for rspconfig to set/get hostname of the BMC

whowutwut opened the pull request:
Last commit Resolves #3295 

The hostname is contained in the `network/config` endpoint: 

```
[root@briggs01 mid05tor12cn05]# ./list.sh
+ curl -c cjar -b cjar -k -H 'Content-Type: application/json' -X GET https://172.11.140.5/xyz/openbmc_project/network/enumerate
{
  "data": {
    "/xyz/openbmc_project/network/config": {
      "DefaultGateway": "172.11.254.2",
      "HostName": "witherspoon"
    }
```

In this PR, I've added support for GET and SET of hostname using `rspconfig`.  

Externals:
```
rspconfig <noderange> hostname
rspconfig <noderange> hostname=f6u5-bmc
```

Here's what it looks like when executed: 



### GET
```
[root@briggs01 mid05tor12cn05]# rspconfig node-70e28414280d hostname
node-70e28414280d: BMC Hostname: witherspoon
```

On BMC: 
```
# cat /etc/hostname
witherspoon
```

### SET
```
[root@briggs01 mid05tor12cn05]# rspconfig node-70e28414280d hostname=abc123
node-70e28414280d: Setting BMC Hostname (requires bmcreboot to take effect)...
node-70e28414280d: BMC Hostname: abc123
```

On BMC:
```
# cat /etc/hostname
abc123
```

zet809 commented:
Looks good to me, but @whowutwut maybe you need to close and reopen this PR to re-trigger the CI.

whowutwut commented:
* Thanks @gurevichmark !  Updated man pages 
* Changed the external to sync up the output,

With the last commit, here's what it looks like now. 

```
[root@briggs01 xcat]# rspconfig node-70e28414280d hostname
node-70e28414280d BMC: Hostname: witherspoon
[root@briggs01 xcat]# rspconfig node-70e28414280d ip vlan hostname
node-70e28414280d BMC: IP: 172.11.140.5
node-70e28414280d BMC: VLAN ID enabled: n/a
node-70e28414280d BMC: Hostname: witherspoon
[root@briggs01 xcat]# rspconfig node-70e28414280d hostname=abc123
node-70e28414280d BMC: Setting Hostname (requires bmcreboot to take effect)...
node-70e28414280d BMC: Hostname: abc123
```

whowutwut commented:
I wanted to create a pull request for the last commit separately, but there was too much changes in the same place.  So I exposed the function that we have already tested and is ready for FVT and Customer use regarding rspconfig. 

Here's example of allowing certain options through:
```
[root@briggs01 xcat]# rspconfig node-70e28414280d hostname=xyz123
node-70e28414280d BMC: Setting Hostname (requires bmcreboot to take effect)...
node-70e28414280d BMC: Hostname: xyz123
[root@briggs01 xcat]# rspconfig node-70e28414280d hostname
node-70e28414280d BMC: Hostname: xyz123
[root@briggs01 xcat]# rspconfig node-70e28414280d ip
This openbmc related function is not yet supported. Please contact xCAT development team.
[root@briggs01 xcat]# rspconfig node-70e28414280d vlan
This openbmc related function is not yet supported. Please contact xCAT development team.
[root@briggs01 xcat]# rspconfig node-70e28414280d ip vlan netmask hostname
This openbmc related function is not yet supported. Please contact xCAT development team.
[root@briggs01 xcat]# rspconfig node-70e28414280d ip vlan netmask
This openbmc related function is not yet supported. Please contact xCAT development team.
[root@briggs01 xcat]# rspconfig node-70e28414280d hostname
node-70e28414280d BMC: Hostname: xyz123
[root@briggs01 xcat]# rspconfig node-70e28414280d sshcfg
node-70e28414280d: ssh keys copied to 172.12.253.119
```

whowutwut commented:
@zet809 @gurevichmark Changes are complete, please help review again.

zet809 commented:
hi, @whowutwut , why do you change the output from "node-70e28414280d: BMC Hostname: witherspoon" to "node-70e28414280d BMC: Hostname: witherspoon"?
For our Minsky server or x86 servers, the output looks like this:
```
[root@fs1 ~]# rspconfig frame45cn02 ip netmask gateway
frame45cn02: BMC IP: 50.45.2.1
frame45cn02: BMC Netmask: 255.0.0.0
frame45cn02: BMC Gateway: 50.3.29.1
```

gurevichmark commented:
I think `Usage.pm` still needs to be updated.

whowutwut commented:
@zet809 The reason is that our rspconfig for OpenBMC looks like this for `ip=dhcp` 
```
[root@briggs01 tools]# rspconfig node-70e28414280d ip=dhcp
node-70e28414280d BMC: Setting IP to DHCP...
node-70e28414280d BMC: reboot
```

I was trying to make it consistent with that.   

Independent on how it was implemented in IPMI, we can discuss how it should be implemented moving forward.  If we like the old output, then sure, I can keep it the same ...   but my thought was that `rspconfig` is running actions on the BMC, so is it better to indicate it as the "node" that it's running against.   

Some more examples, regarding bmcstate and state... 
```
[root@briggs01 tools]# rpower mid05tor12cn11 bmcstate
mid05tor12cn11 BMC: Ready
[root@briggs01 tools]# rpower mid05tor12cn11 state
mid05tor12cn11: on

```

Let me know what you think...  others, feel free to chime in as well. 

Maybe a lower case bmc...
```
cn01:
cn01 bmc:
```

whowutwut commented:
@zet809  I changed the externals to match what we have in the IPMI support.  Here's some of the external examples: 

### rspconfig
```
[root@briggs01 ~]# rspconfig  node-8335-gtc-1318c4a hostname
node-8335-gtc-1318c4a: BMC Hostname: xyz123
[root@briggs01 ~]# rspconfig  node-8335-gtc-1318c4a ip vlan
This openbmc related function is not yet supported. Please contact xCAT development team.
```

### rpower 
```
[root@briggs01 ~]# rpower node-8335-gtc-1318c4a state
node-8335-gtc-1318c4a: off
[root@briggs01 ~]# rpower node-8335-gtc-1318c4a bmcstate
node-8335-gtc-1318c4a: BMC Ready
[root@briggs01 ~]# rpower node-8335-gtc-1318c4a bmcreboot
node-8335-gtc-1318c4a: BMC reboot
```

Ready to review again

zet809 commented:
Looks good to me, I will merge, thx @whowutwut
